sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/TransactionalBatch.java

```diff
@@ -0,0 +1,320 @@
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
+
+/**
+ * Represents a batch of operations against items with the same {@link PartitionKey} in a container that will be performed
+ * in a transactional manner at the Azure Cosmos DB service.
+ * <p>
+ * Use {@link TransactionalBatch#createTransactionalBatch(PartitionKey)} or new {@link #TransactionalBatch(PartitionKey)}
+ * to create an instance of TransactionalBatch
+ * <b>Example</b>
+ * This example atomically modifies a set of documents as a batch.
+ * <pre>{@code
+ * public class ToDoActivity {
+ *     public final String type;
+ *     public final String id;
+ *     public final String status;
+ *     public ToDoActivity(String type, String id, String status) {
+ *         this.type = type;
+ *         this.id = id;
+ *         this.status = status;
+ *     }
+ * }
+ *
+ * String activityType = "personal";
+ *
+ * ToDoActivity test1 = new ToDoActivity(activityType, "learning", "ToBeDone");
+ * ToDoActivity test2 = new ToDoActivity(activityType, "shopping", "Done");
+ * ToDoActivity test3 = new ToDoActivity(activityType, "swimming", "ToBeDone");
+ *
+ * TransactionalBatch batch = TransactionalBatch.createTransactionalBatch(new Cosmos.PartitionKey(activityType))
+ *     .createItem<ToDoActivity>(test1)
+ *     .replaceItem<ToDoActivity>(test2.id, test2)
+ *     .upsertItem<ToDoActivity>(test3)
+ *     .deleteItem("reading");
+ *
+ * try (TransactionalBatchResponse response = container.executeTransactionalBatch(batch) {
+ *
+ *     if (!response.IsSuccessStatusCode) {
+ *        // Handle and log exception
+ *        return;
+ *     }
+ *
+ *     // Look up interested results - e.g., via typed access on operation results
+ *
+ *     TransactionalBatchOperationResult<ToDoActivity> result = response.getOperationResultAtIndex<ToDoActivity>(0, ToDoActivity.class);
+ *     ToDoActivity readActivity = result.getItem();
+ * }
+ * }</pre>
+ *
+ * <b>Example</b>
+ * <p>This example atomically reads a set of documents as a batch.
+ * <pre>{@code
+ * String activityType = "personal";
+ *
+ * TransactionalBatch batch = TransactionalBatch.createTransactionalBatch(new Cosmos.PartitionKey(activityType))
+ *     .readItem("playing")
+ *     .readItem("walking")
+ *     .readItem("jogging")
+ *     .readItem("running")
+ *
+ * try (TransactionalBatchResponse response = container.executeTransactionalBatch(batch) {
+ *
+ *     // Look up interested results - eg. via direct access to operation result stream
+ *
+ *     List<String> resultItems = new ArrayList<String>();
+ *
+ *     for (TransactionalBatchOperationResult result : response) {
+ *         resultItems.add(result.getResourceObject().toString())
+ *     }
+ * }
+ * }</pre>
+ * <p>
+ * <b>See:</b>
+ * <a href="https://docs.microsoft.com/azure/cosmos-db/concepts-limits">Limits on TransactionalBatch requests</a>.
+ */
+public final class TransactionalBatch {
```

Comment:
The class and all public members should have the 4_7_0 Beta tag as well

Reply:
Oh. Fixed. Thanks!